src/components/field-label/field-label.tsx

```diff
@@ -14,11 +14,11 @@ type FieldLabelProps = {
   hint: React.ReactNode;
   description: React.ReactNode;
   onInfoButtonClick?: () => void;
-  hintIcon: React.ReactNode;
+  hintIcon: React.ReactElement<any>;
   badge: React.ReactNode;
   hasRequiredIndicator?: boolean;
   htmlFor?: string;
-  horizontalConstraint?: HorizontalConstraintOfFieldLabel;
+  horizontalConstraint: HorizontalConstraintOfFieldLabel;
```

Comment:
That's actually interesting: a prop which is defaulted via `defaultProps` is not optional in sense of typing (makes sense). However most linters won't catch this (e.g. xo does). Generally, when you use the FunctionalComponent annotation, you're stating that your component has an optional defaultProps of type Partial<P>.